src/SecureStore.BeyondTrust/UiPath.Orchestrator.BeyondTrust-DynamicSystem-ReadOnly/BeyondTrustDynamicSystemSecureStore.cs

```diff
@@ -0,0 +1,144 @@
+                    IsMandatory = true,
+                    DefaultValue = "/",
+                },
+                new ConfigurationValue(ConfigurationValueType.Choice)
+                {
+                    Key = "ManagedAccountType",
+                    DisplayName = "Managed Account Type",
+                    IsMandatory = true,
+                    PossibleValues = new List<string> { "system", "domainlinked" }
+                },
+            };
+
+        public SecureStoreInfo GetStoreInfo()
+        {
+            return new SecureStoreInfo { Identifier = "BeyondTrust-DynamicSystem-ReadOnly", IsReadOnly = true };
+        }
+
+        public async Task<string> CreateCredentialsAsync(string context, string key, Credential value)
+        {
+            throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "This secure store is read-only");
+        }
+
+        public async Task<Credential> GetCredentialsAsync(string context, string key)
+        {
+            var config = JsonConvert.DeserializeObject<Dictionary<string, object>>(context);
+            var splitDelimiter = new string[] { config["SystemAccountDelimiter"].ToString() };
+            var keyPieces = key.Split(splitDelimiter, StringSplitOptions.None);
+
+            if (keyPieces.Length > 2)
+            {
+                throw new SecureStoreException(SecureStoreException.Type.InvalidConfiguration, "Splitting by delimiter " + splitDelimiter + " results in " + keyPieces.Length.ToString() + " parts");
+            }
+
+            var managedSystemName = keyPieces[0];
+            var managedAccountName = keyPieces[1];
+
+            var client = BeyondTrustVaultClientFactory.GetClient(context);
+            client.SignIn();
+            var managedAccountResult = client.ManagedAccounts.GetRequestable(managedSystemName, managedAccountName, config["ManagedAccountType"].ToString());
+            if (!managedAccountResult.IsSuccess)
+            {
+                if (managedAccountResult.StatusCode.Equals(HttpStatusCode.NotFound))
+                {
+                    throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "Managed Account not found");
+                }
+                else
+                {
+                    throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "Managed account retreival failed");
+                }
+            }
+
+            var isaRequestResult = client.ISARequests.Post(managedAccountResult.Value.AccountId, managedAccountResult.Value.SystemId, 1, "UiPath Credential Store request");
+            if (!isaRequestResult.IsSuccess)
+            {
+                client.SignOut();
+                throw new SecureStoreException(SecureStoreException.Type.UnsupportedOperation, "ISA request failed with code " + isaRequestResult.StatusCode + " and message: " + isaRequestResult.Message);
+            }
+            client.SignOut();
```

Comment:
You should put `client.SignOut();` in a finally clause

Reply:
Fixed